### ISSUE TITLE
AtomMapping correspondence lookups

### ISSUE THREAD

**Issue by richardjgowers** [OpenFreeEnergy/openfe#8]
depends on #20

**Comment by richardjgowers** [comment]
I'm going to close this as this is technically achieved with `AtomMapping.mol1_to_mol2`, but we might revisit and make this easier/better once we start using it in anger

**Comment by richardjgowers** [comment]
I think something chemical might be something like:
```python
class AtomMapping:
    def __getitem__(self, atom: Atom) -> Union[Atom, None]:
        # ???
        return other_atom
```
But this is a lot harder to make work than it initially seems.  I think sticking to indices to a canonically ordered molecule where possible probably saves a lot of headaches.  As before, I'm also waiting on seeing how awkward future tasks are before loading up AtomMapping with too much functionality.

**Comment by dwhswenson** [comment]
Yep, I was thinking exactly along the lines @richardjgowers proposes. I also get the issues from a practical standpoint. However, isn't all of this implicitly required for @mikemhenry's visualization?

Obviously we haven't yet defined where the `Atom` type comes from, which would be a requirement for this.

**Comment by dwhswenson** [comment]
(This is the main reason I'm pausing before moving this to "done" -- if we get that extra functionality for free with Mike's work-in-progress, then maybe it's worth marking that work as solving either this issue or a separate issue that is specifically defined for that.)
